php-checks/src/main/java/org/sonar/php/checks/UselessIncrementCheck.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import java.util.Deque;
+import java.util.LinkedList;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.php.checks.utils.Equality;
+import org.sonar.php.tree.symbols.Scope;
+import org.sonar.plugins.php.api.symbols.Symbol;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.FunctionDeclarationTree;
+import org.sonar.plugins.php.api.tree.expression.AssignmentExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.UnaryExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.VariableIdentifierTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.ReturnStatementTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = UselessIncrementCheck.KEY)
+public class UselessIncrementCheck extends PHPVisitorCheck {
+
+  public static final String KEY = "S2123";
+  private Deque<Scope> scopes = new LinkedList<>();
```

Comment:
I saw some checks cleaning its state to prevent invalid data cause by a previous exception like this:
```
 @Override
  public void visitCompilationUnit(CompilationUnitTree tree) {
    scopes.clear();
    super.visitCompilationUnit(tree);
  }
```